camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -410,6 +424,7 @@ public String toString() {
         }
     }
 
+    @DslAccessible
```

Comment:
(i.e. I thought the difference was that e.g. `attributeWhenReady()` and `config()` couldn't be invoked within `BrooklynDslInterpreter.evaluateOn()` using `DslDeferredFunctionCall.invoke()`  and so for them a `DslDeferredFunctionCall` object gets created, whereas general `DslComponents` get evaluated with an immediate `DslDeferredFunctionCall.invoke()`.   Now, however, both general `DslComponent`s like `parent()` and also e.g. `attributeWhenReady()/config()` are marked as `@DslAccessible`.